app/src/main/java/com/framgia/quangtran/music_42/mediaplayer/MediaPlayerManager.java

```diff
@@ -0,0 +1,183 @@
+package com.framgia.quangtran.music_42.mediaplayer;
+
+import android.content.Context;
+import android.media.MediaPlayer;
+import android.net.Uri;
+
+import com.framgia.quangtran.music_42.data.model.Track;
+
+import java.io.IOException;
+import java.util.List;
+
+public class MediaPlayerManager extends MediaPlayerSetting implements MediaPlayerInterface
+        , MediaPlayer.OnPreparedListener, MediaPlayer.OnCompletionListener,
+        MediaPlayer.OnErrorListener {
+    private List<Track> mTracks;
+    private static MediaPlayerManager sInstance;
+    private Context mContext;
+    private MediaPlayer mMediaPlayer;
+    private int mTrackCurrentPosition;
+    private int mStatus;
+    protected Exception mException;
+
+    private MediaPlayerManager(Context context) {
+        super();
+        mContext = context;
+    }
+
+    public static MediaPlayerManager getInstance(Context context) {
+        if (sInstance == null) {
+            sInstance = new MediaPlayerManager(context);
+        }
+        return sInstance;
+    }
+
+    @Override
+    public void initMediaPlayer() {
+        if (mMediaPlayer != null && getStatus() != MediaPlayerStates.RELEASE) {
+            mMediaPlayer.reset();
+        } else {
+            mMediaPlayer = new MediaPlayer();
+        }
+    }
+
+    @Override
+    public void initPlay(int position) {
+        if (!mTracks.isEmpty() && position >= 0) {
+            Uri uri = Uri.parse(mTracks.get(position).getStreamUrl());
+            try {
+                mMediaPlayer.setDataSource(mContext, uri);
+                mMediaPlayer.prepare();
+                setStatus(MediaPlayerStates.PREPARING);
+            } catch (IOException e) {
+                mException = e;
+            }
+        }
+    }
+
+    @Override
+    public void start() {
+        mMediaPlayer.start();
+        setStatus(MediaPlayerStates.PLAYING);
```

Comment:
```suggestion
        setState(MediaPlayerStates.PLAYING);
```
Tương tự ở phương thức khác.
Cho anh hỏi chỗ em dùng getState() ở đâu????